src/atomate2/cp2k/sets/core.py

```diff
@@ -0,0 +1,309 @@
+"""Module defining core CP2K input set generators."""
+
+from __future__ import annotations
+
+import logging
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import Any
+
+import numpy as np
+from pymatgen.core import Structure
+from pymatgen.core.periodic_table import Element
+from pymatgen.io.cp2k.inputs import Cp2kInput
+from pymatgen.io.cp2k.outputs import Cp2kOutput
+from pymatgen.io.cp2k.utils import get_truncated_coulomb_cutoff
+
+from atomate2.common.schemas.math import Vector3D
+from atomate2.cp2k.sets.base import Cp2kInputGenerator, multiple_input_updators
+
+logger = logging.getLogger(__name__)
+
+
+__all__ = [
+    "StaticSetGenerator",
+    "RelaxSetGenerator",
+    "CellOptSetGenerator",
+    "HybridStaticSetGenerator",
+    "HybridRelaxSetGenerator",
+    "HybridCellOptSetGenerator",
+    "NonSCFSetGenerator",
+]
+
+
+@dataclass
+class StaticSetGenerator(Cp2kInputGenerator):
+    """
+    Class to generate CP2K static input sets.
+
+    Parameters
+    ----------
+
+    """
+
+    def get_input_updates(self, *args, **kwargs) -> dict:
+        """
+        Get updates to the input for a static CP2K job.
+
+        Parameters
+        ----------
+
+
+        Returns
+        -------
+        dict
+            A dictionary of updates to apply.
+        """
+        updates = {"run_type": "ENERGY_FORCE"}
+        return updates
+
+@dataclass
+class RelaxSetGenerator(Cp2kInputGenerator):
+    """
+
+    """
+
+    def get_input_updates(self, *args, **kwargs) -> dict:
+        """
+        """
+        updates = {
+            "run_type": "GEO_OPT",
+            "activate_motion": {
+                'optimizer': "BFGS",
+                "trust_radius": 0.1
+            },
+        }
+        return updates
+
+@dataclass
+class CellOptSetGenerator(Cp2kInputGenerator):
+    """
+
+    """
+
+    def get_input_updates(self, *args, **kwargs) -> dict:
+        """
+        """
+        updates = {
+            "run_type": "CELL_OPT",
+            "activate_motion": {
+                'optimizer': "BFGS",
+                "trust_radius": 0.1
+            },
+        }
+
+        return updates
+
+
+@dataclass
+class HybridSetGenerator(Cp2kInputGenerator):
+
+    hybrid_functional: str = "PBE0"
+    screen_on_initial_p: bool = False
+    screen_p_forces: bool = False
+    eps_schwarz: float = 1e-7
+    eps_schwarz_forces: float = 1e-5
```

Comment:
Is there a benefit to defining these as class variables? It could be cleaner to put the values in the update function directly.

This should avoid confusion about how you change the hybrid functional. For example, as it stands the hybrid functional of some input set generators can be switched by changing the `hybrid_functional` dataclass field whereas for others you have to set `user_input_settings={"hybrid_functional": "HSE06"}`.

If you remove these class variables (which are a 1:1 mapping of the input update settings anyway) then the only standardised way of changing the functional will be through `user_input_settings`.

Reply:
I think you're right that it should be changed. It's slightly more obtuse for users, because you can see that in order to set hybrid settings, there's this extra layer of having to know that you need to enter a new cp2k module using `{activate_hybrid: {...}}` instead of specifying `hybrid_functional` directly. I'm sure people can look into the source for an example otherwise, and since the hybrid job objects have class variables for setting it, I think you're ultimately right.


Will be resolved in next commit.